### PR TITLE
Asset ability component

### PR DESCRIPTION
**Pull request by rsek**
Based on #684 

Adds an `IronCheckbox`-based asset ability component, which is now used by all asset components. It has various props to configure the desired level of interactivity.

Currently the bullet shapes are set via computed property, but in the future this will be configured via CSS variables or design tokens.

### CODE REVIEW

File: src/module/vue/components/asset/asset-ability.vue [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+					rotate: FontAwesome.Rotate['90deg'],
+					family: FontAwesome.Family.Solid
+				}
+			}
+		default:
+			return {
+				icon: 'fa:circle',
+				props: { family: FontAwesome.Family.Solid }
+			}
+	}
```
Comment (ben) [inline src/module/vue/components/asset/asset-ability.vue:73]: This is a weird way to do this:

```suggestion
	if (IronswornSettings.starforgedToolsEnabled) {
		return {
			icon: 'fa:hexagon',
			props: {
				rotate: FontAwesome.Rotate['90deg'],
				family: FontAwesome.Family.Solid
			}
		}
	}
	return {
		icon: 'fa:circle',
		props: { family: FontAwesome.Family.Solid }
	}
```

**Review by ben — COMMENTED** [review]
I like it! Just one question about flow-control code style.

File: src/module/vue/components/asset/asset-ability.vue (hunk 1 above)
Comment (rsek) [reply src/module/vue/components/asset/asset-ability.vue:73]: yeah, fair. i think my reasoning (can't be sure, it was pretty late) was that it'd be easier to add additional options to a switch, but that'd probably be handled in a totally different theme file anyways.

File: src/module/vue/components/asset/asset-ability.vue [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+			}
+		default:
+			return {
+				icon: 'fa:circle',
+				props: {
+					family: FontAwesome.Family.Regular
+				}
+			}
+	}
+})
```
Comment (ben) [inline src/module/vue/components/asset/asset-ability.vue:91]: You also did it here 😁

**Pull request merged**